garage_0_8: 0.8.4 -> 0.8.5, garage_0_9: 0.9.0 -> 0.9.1

adamcstephens opened the pull request:
## Description of changes

https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.8.5
https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.9.1

- garage_0_8: 0.8.4 -> 0.8.5
- garage_0_9: 0.9.0 -> 0.9.1
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

adamcstephens committed:
garage_0_8: 0.8.4 -> 0.8.5

https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.8.5

adamcstephens committed:
garage_0_9: 0.9.0 -> 0.9.1

https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.9.1

Due to a recent change in the build system, we have to remove actively a
file (`.cargo/config.toml`) to avoid the mold linker to leak during the linkage phase.

Not sure if we should have more sophisticated ways to ignore this.

adamcstephens committed:
garage: add changelog meta attribute

RaitoBezarius commented:
Result of `nixpkgs-review pr 281572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_1)</li>
    <li>garage_0_8 (garage_0_8_5)</li>
  </ul>
</details>

RaitoBezarius commented:
Result of `nixpkgs-review pr 281572` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_1)</li>
    <li>garage_0_8 (garage_0_8_5)</li>
  </ul>
</details>

adamcstephens committed:
garage_0_8: mark broken on darwin

upstream did not backport fixes for darwin build failures to 0.8.
users should be encouraged to upgrade.

adamcstephens commented:
I marked 0.8 as broken on Darwin. If someone desired they could dig up all the patches and get them to apply, but I'd prefer to encourage people to upgrade instead.

Result of `nixpkgs-review pr 281572` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>garage_0_8 (garage_0_8_5)</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_1)</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 281572` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>garage_0_8 (garage_0_8_5)</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_1)</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 281572` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_1)</li>
    <li>garage_0_8 (garage_0_8_5)</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 281572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_1)</li>
    <li>garage_0_8 (garage_0_8_5)</li>
  </ul>
</details>

NickCao reviewed (approved):
LGTM, merging in two days.
